packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx

```diff
@@ -299,19 +299,28 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
     }
   };
 
-  private _onGotoToday = (): void => {
+  private _onGotoToday = (selectDateOnClick: boolean): void => {
     const { dateRangeType, firstDayOfWeek, today, workWeekDays } = this.props;
 
-    const dates = getDateRangeArray(today!, dateRangeType!, firstDayOfWeek!, workWeekDays!);
+    if (selectDateOnClick) {
+      // When using Defaultprops, TypeScript doesn't know that React is going to inject defaults
+      // so we use exclamation mark as a hint to the type checker (see link below)
+      // https://decembersoft.com/posts/error-ts2532-optional-react-component-props-in-typescript/
+      const dates = getDateRangeArray(today!, dateRangeType!, firstDayOfWeek!, workWeekDays!);
+      this._onSelectDate(today!, dates);
+    }
 
-    this._onSelectDate(today!, dates);
     this._navigateDayPickerDay(today!);
   };
 
+  private _onGotoTodayClick = (ev: React.MouseEvent<HTMLElement>): void => {
+    this._onGotoToday(false /*selectDateOnClick*/);
+  };
+
   private _onGotoTodayKeyDown = (ev: React.KeyboardEvent<HTMLElement>): void => {
     if (ev.which === KeyCodes.enter) {
       ev.preventDefault();
-      this._onGotoToday();
+      this._onGotoToday(false /*selectDateOnClick*/);
```

Comment:
Is onGotoToday ever called with true? I was unclear on why the prop was needed.

There aren't unit tests validating this. Possible to add some?